pallets/cf-reward/src/lib.rs

```diff
@@ -0,0 +1,208 @@
+    /// the maximum lock-time.
+    type LeasePeriod: Get<Self::BlockNumber>;
+    /// Period of vesting for every contributor
+    type VestingPeriod: Get<Self::BlockNumber>;
+    /// This module emits events, and hence, depends on the runtime's definition of event
+    type Event: From<Event<Self>> + Into<<Self as frame_system::Config>::Event>;
+    /// The balance type of the relay chain
+    type RelayChainBalance: Parameter + Member + AtLeast32BitUnsigned + Codec + Default + Copy +
+        MaybeSerializeDeserialize + Debug +
+        Into<<<Self as pallet_vesting::Config>::Currency as Currency<<Self as frame_system::Config>::AccountId>>::Balance>;
+    /// AccountId of the relay chain
+    type RelayChainAccountId: Parameter + Member + MaybeSerializeDeserialize + Debug + MaybeSerialize + Ord
+        + Default;
+}
+
+type BalanceOf<T> = <<T as pallet_vesting::Config>::Currency as Currency<<T as frame_system::Config>::AccountId>>::Balance;
+
+const MODULE_ID: ModuleId = ModuleId(*b"rewards ");
+
+/// Callable functions (i.e. transaction) weight trait
+pub trait WeightInfo {
+    fn initialize() -> Weight;
+    fn reward() -> Weight;
+}
+
+// Define transaction weights to be used when testing
+pub struct TestWeightInfo;
+impl WeightInfo for TestWeightInfo {
+    fn initialize() -> Weight { 0 }
+    fn reward() -> Weight { 0 }
+}
+
+
+decl_event!{
+    pub enum Event<T> where AccountId = <T as frame_system::Config>::AccountId {
+        /// Event emitted when a reward claim was processed successfully.
+        RewardClaimed(AccountId),
+        /// Event triggered when the reward module is ready to reward contributors
+        RewardModuleInitalized,
+    }
+}
+
+decl_storage! {
+    trait Store for Module<T: Config> as Reward {
+        NextInit get(fn next_init) config(): T::BlockNumber;
+        ModuleAccount get(fn mod_account) : T::AccountId = MODULE_ID.into_account();
+        ConversionRate get(fn conversion_rate) config(): BalanceOf<T>;
+    }
+}
+
+decl_error! {
+	pub enum Error for Module<T: Config> {
+        /// Not enough funds in the pot for paying a reward
+        NotEnoughFunds,
+        /// Initalization before the last lease period is over
+        OngoingLease,
+    }
+}
+
+
+decl_module! {
+    pub struct Module<T: Config> for enum Call where origin: <T as frame_system::Config>::Origin {
+        // Activate errors
+        type Error = Error<T>;
+
+        // Activate events
+        fn deposit_event() = default;
+
+        /// Initialize the module.
+        /// This dispatchable function must be called from a root account, so that other accounts
+        /// can not block the module via a random initialization. Furthermore it is important, that
+        /// the module has enough funds, as a later funding is not possible.
+        // TODO: Add correct weight function
+        #[weight = 10_000]
+        fn initialize(origin, source: T::AccountId, amount: BalanceOf<T>) -> DispatchResult {
```

Comment:
@mustermeiszer I'll add benchmarking features to calculate the weights when integrating with `pallet-crowdloan-claim` pallet.